nixos/modules/system/boot/networkd.nix

```
@@ -67,12 +83,38 @@ let
         (assertInt "CombinedChannels")
         (assertRange "CombinedChannels" 1 4294967295)
         (assertInt "RxBufferSize")
+        (assertInt "RxMiniBufferSize")
+        (assertInt "RxJumboBufferSize")
         (assertInt "TxBufferSize")
+        (assertValueOneOf "RxFlowControl" boolValues)
+        (assertValueOneOf "TxFlowControl" boolValues)
+        (assertValueOneOf "AutoNegotiationFlowControl" boolValues)
+        (assertByteFormat "GenericSegmentOffloadMaxBytes")
+        (assertInt "GenericSegmentOffloadMaxSegments")
+        (assertRange "GenericSegmentOffloadMaxSegments" 1 65535)
       ];
     };
 
     netdev = let
 
+      macvlanChecks = [
+        (assertOnlyFields [
+          "Mode"
+          "SourceMACAddress"
+          "BroadcastMulticastQueueLength"
+        ])
+        (assertValueOneOf "Mode" ["private" "vepa" "bridge" "passthru" "source"])
```

```suggestion
        (assertValueOneOf "Mode" ["private" "vepa" "bridge" "passthru" "source"])
        (assertInt "BroadcastMulticastQueueLength")
        (assertRange "BroadcastMulticastQueueLength" 0 4294967294)
```

regarding https://github.com/NixOS/nixpkgs/pull/154207/files#r780910837